js/components/geolocation.js

```diff
@@ -188,6 +192,8 @@ Fliplet.FormBuilder.field('geolocation', {
     value: function(val) {
       if (this.value) {
         this.firstTimeSaved = true;
+      } else {
+        this.firstTimeSaved = false;
```

Comment:
You can skip the `if` entirely

```suggestion
      this.firstTimeSaved = !!this.value;
```